src/tcp.c

```diff
@@ -766,11 +778,90 @@ tcp_server_create
   return ts;
 }
 
+#if ENABLE_LIBSYSTEMD_DAEMON
+
+inline static int
+ip6_addr_cmp(unsigned char a1[16], unsigned char a2[16])
+{
+  int i;
+  int ret = 0;
+
+  for (i = 0; i < 16 && !ret; i++)
+    if (a1[i] != a2[i])
+      ret = 1;
+  return ret;
+}
 
 /**
  *
  */
+void *
+tcp_server_create
+  (const char *bindaddr, int port, tcp_server_ops_t *ops, void *opaque)
+{
+  int sd_fds_num, i, fd;
+  struct sockaddr_storage bound;
+  tcp_server_t *ts;
+  struct in_addr addr4;
+  struct in6_addr addr6;
+  int found = 0;
+
+  sd_fds_num = sd_listen_fds(0);
```

Comment:
Uff.. I just wonder why these three lines and the bellow for loop are not in the systemd library / API. It looks terrible to implement this stuff for all required apps. Have you proposed to include this code to systemd ?

Something like: sd_listen_fds2(bindaddr, port) or so ?